src/library_glfw.js

```diff
@@ -671,6 +672,64 @@ var LibraryGLFW = {
       win.scrollFunc = cbfun;
     },
 
+    setDropCallback: function(winid, cbfun) {
+      var win = GLFW.WindowFromId(winid);
+      if (!win) return;
+      win.dropFunc = cbfun;
+    },
+
+    onDrop: function(event) {
+      if (!GLFW.active.dropFunc) return;
+
+      event.preventDefault();
+
+      var filenames = allocate(new Array(event.dataTransfer.files.length*4), 'i8*', ALLOC_NORMAL);
+      var filenamesArray = [];
+      var count = event.dataTransfer.files.length;
+
+      // Read and save the files to emscripten's FS
+      // TODO: lazily load?
+      var written = 0;
+      for (var i = 0; i < count; ++i) {
+        (function(file) {
+          var reader = new FileReader();
+          reader.onload = function(e) {
```

Comment:
Changed to use [loadend](https://developer.mozilla.org/en-US/docs/Web/Events/loadend), which encompasses load/error/abort, and checked the readyState.